### ISSUE TITLE
Change `filter` front matter key to `filterOut` for v2.x

### ISSUE THREAD

**Issue by Zearin** [11ty/eleventy#2522]
**Is your feature request related to a problem? Please describe.**
When writing a `.filter( … ) ` function in JavaScript, I tend to think of it as “Filter FOR «truthy condition»”. In other words, I think in terms of what I want to _include_.

But the [`filter` front matter key](https://www.11ty.dev/docs/pagination/#filtering-values) filters based on what you want to _exclude_.

**Describe the solution you'd like**
For more human-friendliness, I recommend changing the name of this key to `filterOut` for the official v2.x release of Eleventy.

**Describe alternatives you've considered**
Optionally, you could allow Eleventy to continue support for `filter` with a deprecation warning.

(You could also keep the `filter` keyword, and flip the logic that Eleventy uses…but that is hostile to users that have become used to the current behavior.)

**Comment by mayankkamboj47** [comment]
Maybe we could use ‘exclude’ instead of ‘filterOut’ if that’s not already taken

**Comment by Snapstromegon** [comment]
I agree with @mayankkamboj47 to choose 'exclude' over 'filterOut'. Not only because it seems more semantic to me, but it also allows for the addition of an 'include' property which might filter what to include from the collection.

**Comment by Zearin** [comment]
@zachleat Please don’t forget this before v2.0 leaves beta! 🙏

**Comment by uncenter** [comment]
I'm all for adding `include` and `exclude` properties and deprecating `filter`.

**Comment by Snapstromegon** [comment]
Especially @zachleat:
I'm willing to create a PR for this, but how would you expect it to work when both `include` and `exclude` are specified?

I personally see these options:

**1. Disallow**
Just throw/warn/whatever when both keys are specified

**2. Exclude over Include**
Exclude items even when they are included.

**3. Include over Exclude**
Include items even when they are excluded.

I'd personally tend to use option 1, but I'm completely open to suggestions.

**Comment by uncenter** [comment]
Option 1 is the only one that makes sense to me, I'd go with that.

**Comment by Zearin** [comment]
Maybe this can squeeze in before the final release of v3?

**Comment by zachleat** [comment]
Ruthlessly punting this one to 4.0, sorry!

**Comment by Zearin** [comment]
😭 

J/k. I'm good. 🙂
